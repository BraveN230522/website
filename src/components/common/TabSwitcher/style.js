import styled from '@emotion/styled';

const Tabs = styled.div`
  display: flex;
  width: 100%;
  height: 60px;

  button {
    &:nth-of-type(1) {
      border-top-left-radius: 16px;
    }
    &:nth-of-type(2) {
      border-top-right-radius: 16px;
    }
    ${({ theme }) => theme.media.phone`
    &:nth-of-type(1) {
      border-top-left-radius: 0;
    }
    &:nth-of-type(2) {
      border-top-right-radius: 0;
    }
  `};
  }
`;

const TabButton = styled.button`
  display: flex;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  font-family: ${({ theme }) => theme.fonts.DINAlternate};
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  text-decoration: none;
  border: none;

  ${({ theme, active }) => `
    background-color: ${active ? theme.colors.white : theme.colors.lightStone};
    color: ${active ? theme.colors.black : theme.colors.browGray};
  `};

  &:hover {
    text-decoration: none;
    cursor: pointer;
    border: none;
  }

  &:focus {
    outline: none;
    ${({ active }) =>
      !active &&
      `
      box-sizing: border-box;
      border: none;
    `};
  }
  ::-moz-focus-inner {
    border: 0;
  }
`;

export { Tabs, TabButton };
