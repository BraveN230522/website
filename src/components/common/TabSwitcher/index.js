import React, { useEffect, useCallback, useRef } from 'react';

import { Tabs, TabButton } from './style';

const TabSwitcher = ({ leftLabel, rightLabel, active, activeCallback, shouldSetInitialFocus }) => {
  const leftRef = useRef();

  useEffect(() => {
    if (leftRef.current && shouldSetInitialFocus) {
      leftRef.current.focus();
    }
  }, [leftRef.current]);

  const handleLeft = useCallback(() => {
    if (active !== leftLabel) {
      activeCallback(leftLabel);
    }
  }, [active]);

  const handleRight = useCallback(() => {
    if (active !== rightLabel) {
      activeCallback(rightLabel);
    }
  }, [active]);

  return (
    <Tabs data-testid="tab-switcher-component">
      <TabButton active={active === leftLabel} onClick={handleLeft} ref={leftRef}>
        {leftLabel}
      </TabButton>
      <TabButton active={active === rightLabel} onClick={handleRight}>
        {rightLabel}
      </TabButton>
    </Tabs>
  );
};

TabSwitcher.defaultProps = {
  shouldSetInitialFocus: false,
};

export default TabSwitcher;
