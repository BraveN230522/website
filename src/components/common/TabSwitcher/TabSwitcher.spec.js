import React, { useState } from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import { matchers } from 'jest-emotion';

import TabSwitcher from './index';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(matchers);

afterEach(() => {
  cleanup();
});

const testId = 'tab-switcher-component';

const WithState = props => {
  const { leftLabel, rightLabel, shouldSetInitialFocus } = props;
  const [active, setActive] = useState('left');

  return (
    <>
      <TabSwitcher
        leftLabel={leftLabel}
        rightLabel={rightLabel}
        active={active}
        activeCallback={setActive}
        shouldSetInitialFocus={shouldSetInitialFocus}
      />
      <span>{`active tab is ${active}`}</span>
    </>
  );
};
WithState.defaultProps = {
  leftLabel: 'left',
  rightLabel: 'right',
  shouldSetInitialFocus: false,
};

const setup = shouldSetInitialFocus => {
  const utils = render(<WithState shouldSetInitialFocus={shouldSetInitialFocus} />, {
    wrapper: withGlobals,
  });

  return utils;
};

describe('<TabSwitcher />', () => {
  it('Renders the TabSwitcher unchanged', () => {
    const { asFragment, queryByTestId } = setup();
    expect(asFragment()).toMatchSnapshot();
    expect(queryByTestId(testId)).toBeTruthy();
  });

  it('Switch between left and right content', () => {
    const { getByText } = setup();

    // initial active tab
    expect(getByText(/active tab is left/i));

    // switch to right
    fireEvent.click(getByText('right'));
    expect(getByText(/active tab is right/i));

    // switch back to left
    fireEvent.click(getByText('left'));
    expect(getByText(/active tab is left/i));
  });

  it('Focus on left if shouldSetInitialFocus is true', () => {
    const shouldSetInitialFocus = true;
    const { getAllByText } = setup(shouldSetInitialFocus);
    expect(document.activeElement).toBe(getAllByText('left')[0]);
  });

  it('Should not focus on left if shouldSetInitialFocus is false', () => {
    const shouldSetInitialFocus = false;
    const { getAllByText } = setup(shouldSetInitialFocus);
    expect(document.activeElement).not.toBe(getAllByText('left')[0]);
  });
});
