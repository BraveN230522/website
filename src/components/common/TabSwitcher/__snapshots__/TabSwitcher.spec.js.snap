// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabSwitcher /> Renders the TabSwitcher unchanged 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 60px;
}

.emotion-4 button:nth-of-type(1) {
  border-top-left-radius: 16px;
}

.emotion-4 button:nth-of-type(2) {
  border-top-right-radius: 16px;
}

@media (max-width:48em) {
  .emotion-4 button:nth-of-type(1) {
    border-top-left-radius: 0;
  }

  .emotion-4 button:nth-of-type(2) {
    border-top-right-radius: 0;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: DINAlternateRegular,sans-serif;
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  background-color: #ffffff;
  color: #000000;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0::-moz-focus-inner {
  border: 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: DINAlternateRegular,sans-serif;
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  background-color: #eeeeee;
  color: #9c9c9c;
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-2:focus {
  outline: none;
  box-sizing: border-box;
  border: 3px solid;
}

.emotion-2::-moz-focus-inner {
  border: 0;
}

<div
    class="emotion-4 emotion-5"
    data-testid="tab-switcher-component"
  >
    <button
      class="emotion-0 emotion-1"
    >
      left
    </button>
    <button
      class="emotion-2 emotion-1"
    >
      right
    </button>
  </div>
  <span>
    active tab is left
  </span>
</DocumentFragment>
`;
